cli/credentials/credentials.go

```diff
@@ -0,0 +1,123 @@
+package credentials
+
+import (
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/yaml"
+	"golang.org/x/net/context"
+	yamlbase "gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+// Credentials contains a set of Username + AccessKey for SauceLabs.
+type Credentials struct {
+	Username  string `yaml:"username"`
+	AccessKey string `yaml:"accessKey"`
+	Source    string
+}
+
+// GetCredentials returns the currently configured credentials (env is prioritary vs. file).
+func GetCredentials() *Credentials {
+	if envCredentials := GetCredentialsFromEnv(); envCredentials != nil {
+		return envCredentials
+	}
+	return GetCredentialsFromFile()
+}
+
+// GetCredentialsFromEnv reads the credentials from the user environment.
+func GetCredentialsFromEnv() *Credentials {
```

Comment:
Redundant because of package name (non canonical go).
```suggestion
func FromEnv() *Credentials {
```